src/client/datascience/notebook/notebookControllerManager.ts

```diff
@@ -283,28 +290,40 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
     private createNotebookController(
         kernelConnection: KernelConnectionMetadata,
         label: string
-    ): VSCodeNotebookController {
-        // Create notebook selector
-        const controller = new VSCodeNotebookController(
-            kernelConnection,
-            label,
-            this.notebook,
-            this.commandManager,
-            this.kernelProvider,
-            this.preferredRemoteKernelIdProvider,
-            this.context,
-            this,
-            this.pathUtils,
-            this.disposables
-        );
-
-        // Hook up to if this NotebookController is selected or de-selected
-        controller.onNotebookControllerSelected(this.handleOnNotebookControllerSelected, this, this.disposables);
-
-        // We are disposing as documents are closed, but do this as well
-        this.disposables.push(controller);
-
-        return controller;
+    ): VSCodeNotebookController | undefined {
+        try {
+            // Create notebook selector
+            const controller = new VSCodeNotebookController(
+                kernelConnection,
+                label,
+                this.notebook,
+                this.commandManager,
+                this.kernelProvider,
+                this.preferredRemoteKernelIdProvider,
+                this.context,
+                this,
+                this.pathUtils,
+                this.disposables
+            );
+
+            // Hook up to if this NotebookController is selected or de-selected
+            controller.onNotebookControllerSelected(this.handleOnNotebookControllerSelected, this, this.disposables);
+
+            // We are disposing as documents are closed, but do this as well
+            this.disposables.push(controller);
+
+            return controller;
+        } catch (ex) {
+            // We know that this fails when we have xeus kernels installed (untill that's resolved thats one instance when we can have duplicates).
+            sendTelemetryEvent(
```

Comment:
If you install the xeus kernel, then everythings falls over. You cannot run any notebook, because we end up with duplicate kernels in our code.
This ensures we never end up in a situtaion where we're completely broken because of one bad apple.
Have a separate issue for handling xeus python kernels.

@IanMatthewHuff @rchiodo

Reply:
Good catch @DonJayamanne I should install that kernel as well just to have for testing.